src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -2157,14 +2157,14 @@ static void update_bdp(grpc_exec_ctx *exec_ctx, grpc_chttp2_transport *t,
 
 // TODO(ncteisen): combine this logic with above func
 static void update_frame(grpc_exec_ctx *exec_ctx, grpc_chttp2_transport *t,
-                         double bw_dbl) {
-  int32_t target = (int32_t)bw_dbl / 1000000;
+                         double bw_dbl, double bdp_dbl) {
+  int32_t bdp = GPR_CLAMP((int32_t)bdp_dbl, 128, INT32_MAX);
+  int32_t target = GPR_MAX((int32_t)bw_dbl / 1000, bdp);
```

Comment:
Max looks right, but the title of the commit is wrong. It should be max, though.